homeassistant/components/auth/mfa_setup_flow.py

```diff
@@ -164,6 +164,6 @@ def _prepare_result_json(
     if (schema := data["data_schema"]) is None:
         data["data_schema"] = []  # type: ignore[typeddict-item]  # json result type
     else:
-        data["data_schema"] = voluptuous_serialize.convert(schema)
+        data["data_schema"] = voluptuous_serialize.convert(schema)  # type: ignore[typeddict-item]  # json result type
```

Comment:
Those functions all take a `FlowResult` and return a `FlowResult`, but change the "data_schema" key of the input FlowResult to a different type. Should there be a second type definition for the returns here?